rpc/src/eth.rs

```diff
@@ -241,6 +241,33 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApi<B, C, P, CT, BE, H> where
 		}
 		Ok(None)
 	}
+
+	fn load_transactions(&self, hash: H256) -> Result<Option<(H256, u32)>> {
+		let mut transactions: Vec<(H256, u32)> = Vec::new();
+		match frontier_consensus::load_transaction_metadata(
+			self.client.as_ref(),
+			hash,
+		).map_err(|err| internal_err(format!("fetch aux store failed: {:?}", err)))? {
+			Some(metadata) => {
+				for (hash, index) in metadata {
+					match self.load_hash(hash)
+						.map_err(|err| internal_err(format!("{:?}", err)))?
+					{
+						Some(_) => {
+							transactions.push((hash, index));
```

Comment:
@tgmichel Am I understanding it correctly this is for the parachain issue? In normal non-parachain best-chain selector, all block hashes should exist (that is, only this `Some` branch would be called, but not the `None` branch).

Reply:
Not related, it's part of the EthApi method `fn load_hash`, that may return None when cannot get the header with a given BlockId or fail to retrieve from the auxstore -> https://github.com/paritytech/frontier/blob/master/rpc/src/eth.rs#L221

Reply:
The thing is I don't know why this is necessary. If we get the `(block_hash, index)` tuple, the `block_hash` is supposed to always exist in the Aux.

Reply:
I see what you mean, yes I guess we can take for granted that the block exists.

Reply:
As we talked, I got it wrong in my own PR!

With `match self.load_hash(block_hash)` we are finding out which block, of the _eventual_ multiple blocks mapped to a given transaction, is part of the canonical chain. So it is required.

I just reset to the previous commit, sorry about this.